source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -361,6 +361,22 @@ std::string SslSocket::subjectLocalCertificate() const {
   return Utility::getSubjectFromCertificate(*cert);
 }
 
+SystemTime SslSocket::validFromPeerCertificate() const {
+  bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl_.get()));
+  if (!cert) {
+    return std::chrono::system_clock::from_time_t(0);
```

Comment:
Perhaps it would be better to use an `absl:optional` here and below instead of returning 0?

Reply:
makes sense to me. I would assume the caller would use `peerCertificatePresented()` first to check that these functions would return something meaningful